### PR TITLE
Added method to get list of logs with optional filtering

### PR DESCRIPTION
**Pull request by fubuloubu**
### - What I did

Added `get_logs()` function to retrieve the list of logs produced by a contract in a given receipt, optional filtering argument to only collect logs of a specific type
Changed `get_log` fixture to use this new function (returning the last log in the filtered list)

### - How to verify it

Tests all work, let me know if there are any concerns

### - Description for the changelog

Added get_logs(), redefined get_log fixture to use get_logs()

### - Cute Animal Picture

turtle

### CODE REVIEW

File: tests/setup_transaction_tests.py [hunk 1]
```diff
@@ -103,26 +103,25 @@ def get_contract(source_code, *args, **kwargs):
 
 curve_order = 21888242871839275222246405745257275088548364400416034343698204186575808495617
 
+def get_logs(receipt, contract, event_name=None):
+    contract_log_ids = contract.translator.event_data.keys() # All the log ids contract has
```
Comment (DavidKnott) [inline tests/setup_transaction_tests.py]: Why does `event_name` default to `None` when it's required in `get_log`?

Comment (jacqueswww) [reply tests/setup_transaction_tests.py]: @DavidKnott @fubuloubu Hmm good point, considering this change. Can we rename get_log to get_last_log() and make `event_name=None` within `get_last_log` this way it's clear we want the last entry in the log (with optional even_name filtering).

Comment (fubuloubu) [reply tests/setup_transaction_tests.py]: `event_name` is a filter, `None` means no filter applied
I like the suggestion about `get_last_log()` but that requires updates to the places it was used.

**Comment by fubuloubu** [comment]
Finally got around to making the suggested changes

**Comment by fubuloubu** [comment]
@DavidKnott any other issues?

**Comment by DavidKnott** [comment]
Nope, LGTM.  Thanks for the contribution!

**Pull request merged**